azure-pipelines/jobs/ci-linux.yml

```diff
@@ -57,18 +58,35 @@ jobs:
         )"
     continueOnError: true
 
+  # Files may be owned by root because builds don't set user ID. Later build steps run 'find' in
+  # the source tree, which fails due to permissions in the 'NetCore*-Int-Pool' queues. This step
+  # prevents the failure by using chown to clean up our source tree.
+  - script: |
+      set -x
+      docker run --rm \
+        -v "$(Agent.BuildDirectory):/root/build" \
+        -w /root/build \
+        bash -c "chown -R $(id -u):$(id -g) *"
+    displayName: Update file ownership from root to build agent account
```

Comment:
It might be useful to add a pinned link for the source of this step to the comment for traceability reasons. (E.g. finding the original PR that invented this, or if there's a bug in the future, seeing if the source location has been changed more recently in response to the same bug.)